Orange/distance/distance.py

```diff
@@ -645,33 +645,31 @@ class PearsonRAbsolute(CorrelationDistance):
     def fit(self, _):
         return PearsonModel(True, self.axis, self.impute)
 
+
 def _prob_dist(a):
     # Makes the vector sum to one, as to mimic probability distribution.
     return a / np.sum(a)
 
-def non_negative(a):
-    #Raise an exception for infinities, nans and negative values
-    try:
-        check_array(a, accept_sparse=True, accept_large_sparse=True, ensure_2d=False)
-    except:
-        raise ValueError("Bhattcharyya distance requires non-negative values")
-    if sp.issparse(a):
-        if a.min() < 0:
-            raise ValueError("Bhattcharyya distance requires non-negative values")
-        return
-    if min(a) < 0:
+
+def check_non_negative(a):
+    # Raise an exception for infinities, nans and negative values
+    check_array(a,
+                accept_sparse=True, accept_large_sparse=True, ensure_2d=False)
+    if a.min() < 0:
```

Comment:
@AndrejaKovacic, would this be OK, too?

Dense arrays also have a method `min`, so there's no need for `if`. Also, I think it is better to not change the exception message raised by `check_array` so the caller is informed that, for instance, there are `nan` values in the data.

Reply:
Yes, it's more informative this way.